toolchain/check/testdata/impl/impl_forall.carbon

```diff
@@ -16,19 +16,23 @@ impl forall [T:! type] T as Simple {
 // CHECK:STDOUT:
 // CHECK:STDOUT: constants {
 // CHECK:STDOUT:   %.1: type = interface_type @Simple [template]
+// CHECK:STDOUT:   %.2: type = assoc_entity_type @Simple, <function> [template]
+// CHECK:STDOUT:   %.3: <associated <function> in Simple> = assoc_entity element0, @Simple.%F [template]
 // CHECK:STDOUT: }
 // CHECK:STDOUT:
 // CHECK:STDOUT: file {
 // CHECK:STDOUT:   package: <namespace> = namespace {.Simple = %Simple.decl} [template]
 // CHECK:STDOUT:   %Simple.decl = interface_decl @Simple, () [template = constants.%.1]
-// CHECK:STDOUT:   impl_decl @impl, (<unexpected instref inst+4>, <unexpected instref inst+5>, <unexpected instref inst+6>, <unexpected instref inst+7>)
+// CHECK:STDOUT:   impl_decl @impl, (<unexpected instref inst+7>, <unexpected instref inst+8>, <unexpected instref inst+9>, <unexpected instref inst+10>)
```

Comment:
I think these "unexpected instref" should be fixed because this is a non-fail test.

Reply:
I agree. Will take a look after this lands (they're not new here or related to this change at all).